architecture/additional_concepts/flannel.adoc

```diff
@@ -29,4 +31,9 @@ responsibile for:
 
 Each *flanneld* agent provides this infomation to a centralized *etcd* store so
 other agents on hosts can create an overlay network and route packets to
-any container within the *flannel* network.
\ No newline at end of file
+any container within the *flannel* network.
+
+The following diagram illustrates the architecture and data flow from one
+container to another using a *flannel* network:
+
+image::flannel.png[Flannel Communication]
```

Comment:
This is a diagram of the default flannel configuration, using an overlay network, but our ansible-based flannel install uses flannel in the "host-gw" mode, which is totally different. (The default config does, technically, avoid "dual Open vSwitch" (since flannel doesn't use OVS) but it doesn't avoid double encapsulation of data, which is the actual problem. The "host-gw" mode uses a completely different architecture to avoid that.